src/Nethermind/Nethermind.Peering.Mothra/MothraLibp2p.cs

```diff
@@ -0,0 +1,240 @@
+        private GCHandle _discoveredPeerHandle;
+        private readonly MothraInterop.ReceiveGossip _receiveGossip;
+        private GCHandle _receiveGossipHandle;
+        private readonly MothraInterop.ReceiveRpc _receiveRpc;
+        private GCHandle _receiveRpcHandle;
+
+        public MothraLibp2p()
+        {
+            unsafe
+            {
+                _discoveredPeer = DiscoveredPeerHandler;
+                _receiveGossip = ReceiveGossipHandler;
+                _receiveRpc = ReceiveRpcHandler;
+            }
+
+            _discoveredPeerHandle = GCHandle.Alloc(_discoveredPeer);
+            _receiveGossipHandle = GCHandle.Alloc(_receiveGossip);
+            _receiveRpcHandle = GCHandle.Alloc(_receiveRpc);
+        }
+
+        public event EventHandler<GossipReceivedEventArgs>? GossipReceived;
+
+        public event EventHandler<PeerDiscoveredEventArgs>? PeerDiscovered;
+
+        public event EventHandler<RpcReceivedEventArgs>? RpcReceived;
+
+        public void SendGossip(string topic, ReadOnlySpan<byte> data)
+        {
+            byte[] topicUtf8 = Encoding.UTF8.GetBytes(topic);
+            unsafe
+            {
+                fixed (byte* topicUtf8Ptr = topicUtf8)
+                fixed (byte* dataPtr = data)
+                {
+                    MothraInterop.SendGossip(topicUtf8Ptr, topicUtf8.Length, dataPtr, data.Length);
+                }
+            }
+        }
+
+        public void SendRpcRequest(string method, string peer, ReadOnlySpan<byte> data)
+        {
+            byte[] methodUtf8 = Encoding.UTF8.GetBytes(method);
+            byte[] peerUtf8 = Encoding.UTF8.GetBytes(peer);
+            unsafe
+            {
+                fixed (byte* methodUtf8Ptr = methodUtf8)
+                fixed (byte* peerUtf8Ptr = peerUtf8)
+                fixed (byte* dataPtr = data)
+                {
+                    MothraInterop.SendRequest(methodUtf8Ptr, methodUtf8.Length, peerUtf8Ptr, peerUtf8.Length, dataPtr,
+                        data.Length);
+                }
+            }
+        }
+
+        public void SendRpcResponse(string method, string peer, ReadOnlySpan<byte> data)
+        {
+            byte[] methodUtf8 = Encoding.UTF8.GetBytes(method);
+            byte[] peerUtf8 = Encoding.UTF8.GetBytes(peer);
+            unsafe
+            {
+                fixed (byte* methodUtf8Ptr = methodUtf8)
+                fixed (byte* peerUtf8Ptr = peerUtf8)
+                fixed (byte* dataPtr = data)
+                {
+                    MothraInterop.SendRequest(methodUtf8Ptr, methodUtf8.Length, peerUtf8Ptr, peerUtf8.Length, dataPtr,
+                        data.Length);
+                }
+            }
+        }
+
+        public void Start(MothraSettings settings)
+        {
+            MothraInterop.RegisterHandlers(_discoveredPeer, _receiveGossip, _receiveRpc);
+            string[] args = BuildArgs(settings);
+            _argsHandle = GCHandle.Alloc(args);
```

Comment:
do we need to release this allocation somehwere?

Reply:
not much memory and one invocation so probably can be ignored

Reply:
Not sure, these are start up args, but I presume (following convention) that we own them and can deallocate after the call... actually I think this is old code from trying to get Mothra working and I may even be able to remove. I will check.

Reply:
Removed. Seems to work fine.